bottles/frontend/views/new.py

```diff
@@ -176,80 +162,119 @@ def set_path(_dialog, response):
             task_func=self.manager.create_bottle,
             callback=self.finish,
             name=self.entry_name.get_text(),
             path=self.custom_path,
-            environment=self.selected_env,
+            environment=self.__radio_get_active(),
             runner=self.runner,
-            arch=arch,
+            arch=list(self.arch)[self.combo_arch.get_selected()],
             dxvk=self.manager.dxvk_available[0],
-            sandbox=sandbox_state,
+            sandbox=self.switch_sandbox.get_state(),
             fn_logger=self.update_output,
             custom_environment=self.env_recipe_path
         )
 
-    def update_output(self, text):
+    def update_output(self, text: str) -> None:
         """
-        This function update the label_output with the given text.
-        It will be concatenated with the previous one.
+        Updates label_output with the given text by concatenating
+        with the previous text.
         """
         current_text = self.label_output.get_text()
         text = f"{current_text}{text}\n"
         self.label_output.set_text(text)
 
-    def finish(self, result, error=None):
+    def finish(self, result, error=None) -> None:
+        """ Updates widgets based on whether it succeeded or failed. """
+
+        def send_notification(notification_id: str) -> None:
+            """ Sends notification if out of focus. """
+            if not self.is_active():
+                self.app.send_notification(notification_id, notification)
+
+        self.status_statuses.set_description(None)
+        self.is_closable = True
+        notification = Gio.Notification()
+
+        # Show error if bottle unsuccessfully builds
         if not result or not result.status or error:
-            self.update_output(_("There was an error creating the bottle."))
+            title = _("Unable to Create Bottle")
             self.btn_cancel.set_visible(False)
             self.btn_close.set_visible(True)
-            self.headerbar.remove_css_class("flat")
+            notification.set_title(title)
+            notification.set_body(_("Bottle failed to create with one or more errors."))
+            self.status_statuses.set_title(title)
+            self.btn_close.get_style_context().add_class("destructive-action")
+            send_notification("bottle-failed")
             return
 
-        self.new_bottle_config = result.data.get("config")
-        self.created.set_description(
-            _("\"{0}\" was created successfully.").format(
+        # Show success
+        title = _("Bottle Created")
+        description = _("\"{0}\" was created successfully.").format(
                 self.entry_name.get_text()
             )
-        )
-        self.btn_cancel.set_visible(False)
-        self.stack_create.set_visible_child_name("page_created")
 
-        '''
-        Ask the manager to check for new bottles, then update the
-        user bottles list. 
-        '''
+        notification = Gio.Notification()
```

Comment:
This redefines `notification`. I would also prefer if this declaration was *before* the `send_notification` function, it's more clear but I can also understand why it's here